ets-demo/etsdemo/main.py

```diff
@@ -10,14 +10,76 @@
 
 """ The main function for launching the demo application.
 """
+import argparse
+import contextlib
+import logging
 
 from etsdemo.app import Demo, DemoVirtualDirectory
 from etsdemo.loader import get_responses, response_to_node
 
 
+#: Logging constants
+_LOG_FORMAT = "%(asctime)s %(levelname)-8.8s [%(name)s:%(lineno)s] %(message)s"
+logger = logging.getLogger(__name__)
+
 #: Default application title
 _TITLE = "Enthought Tool Suite"
 
+_THIS_PACKAGE, _ = __name__.split(".", 1)
+
+
+@contextlib.contextmanager
+def _set_logger(logger, level):
+    """ Context manager for setting logging.
+
+    Parameters
+    ----------
+    logger : logging.Logger
+        Logger to be configured.
+    level : int
+        Logging level to set.
+    """
+    formatter = logging.Formatter(_LOG_FORMAT)
+    handler = logging.StreamHandler()
+    handler.setLevel(level)
+    handler.setFormatter(formatter)
+
+    original_level = logger.level
+    logger.setLevel(level)
+    logger.addHandler(handler)
+    try:
+        yield
+    finally:
+        logger.removeHandler(handler)
+        logger.setLevel(original_level)
+
+
+def _parse_command_line():
+    """ Parse command line arguments for the main function.
+
+    Returns
+    -------
+    namespace : argparse.Namespace
+        Parsed command line arguments.
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--debug',
+        help="Log debugging information to console.",
+        action="store_const",
+        dest="log_level",
+        const=logging.DEBUG,
+        default=logging.WARNING,
```

Comment:
I think i finally understand what's happening here. Not passing `--debug` sets `log_level` to `logging.WARNING` and passing `--debug` sets `log_level` to `logging.DEBUG`

Reply:
That's right.